### DISCUSSION TITLE
Update the libigl example project to include all possible igl modules

### DISCUSSION THREAD

**Discussion by evouga** [libigl/libigl#2284] [General]
It would be helpful if the example project included all possible LibIGL modules and their proper incantations (possibly commented out) so that the user can choose the ones they want to use and delete the rest when cloning the example project.

Currently the template includes only `glfw`. If you want to add, say, Triangle to the project, you run into three issues:
1. You have to know that the module is called `triangle`
2. You have to know to change `igl_include(triangle)` to `igl_include(restricted triangle)`
3. You have to know to change `igl::triangle` to `igl_restricted::triangle`

None of which, as far as I can tell, is documented.

**Comment by alecjacobson** [comment]
I agree. I had some massive "example to rule them all" at some point. I think for now commented out would be ok.

The libigl cmake jungle is not easy to navigate, however, there is this thing in the main repo: [libigl/LibiglOptions.cmake.sample](https://github.com/libigl/libigl/blob/main/LibiglOptions.cmake.sample)

**Reply by alecjacobson** [reply]
@evouga please see if the newest changes here https://github.com/libigl/libigl-example-project/commit/62cfb291a27920f21d21522aa9b92708805ce0c0 are sufficient.

**Reply by evouga** [reply]
Yes, this looks great!

**Comment by alecjacobson** [comment]
btw, you shouldn't have to do step 2 in your original post.

you should only need to add `igl_restricted::triangle` to your `target_link_libraries` command.

**Reply by alecjacobson** [reply]
errr. maybe that's not true. I usually just flip the LIBIGL_RESTRICTED_TRIANGLE on... not sure why there's a `igl_include` indirectrion.

**Reply by alecjacobson** [reply]
So, specifically, I'm not sure why we have:

```cmake
include(libigl)
igl_include(glfw)
```

instead of 

```cmake
option(LIBIGL_RESTRICTED_TRIANGLE "Build target igl_restricted::triangle" ON)
include(libigl)
```

I _think_ they're equivalent, but only @jdumas would know.
